static/app/components/commandPalette/context.tsx

```diff
@@ -0,0 +1,104 @@
+import {createContext, useCallback, useContext, useMemo, useReducer} from 'react';
+
+import type {CommandPaletteAction} from './types';
+
+type CommandPaletteProviderProps = {children: React.ReactNode};
+
+type CommandPaletteStore = {
+  actions: CommandPaletteAction[];
+};
+
+type CommandPaletteConfig = {
+  registerActions: (actions: CommandPaletteAction[]) => void;
+  unregisterActions: (keys: string[]) => void;
+};
+
+type CommandPaletteActionReducerAction =
+  | {
+      actions: CommandPaletteAction[];
+      type: 'REGISTER_ACTIONS';
+    }
+  | {
+      keys: string[];
+      type: 'UNREGISTER_ACTIONS';
+    };
+
+const CommandPaletteConfigContext = createContext<CommandPaletteConfig | null>(null);
+const CommandPaletteStoreContext = createContext<CommandPaletteStore | null>(null);
+
+export function useCommandPaletteConfiguration(): CommandPaletteConfig {
+  const ctx = useContext(CommandPaletteConfigContext);
+  if (ctx === null) {
+    throw new Error('Must be wrapped in CommandPaletteProvider');
```

Comment:
A tad friendlier if it does happen
```suggestion
    throw new Error('useCommandPaletteConfiguration must be wrapped in CommandPaletteProvider');
```